drivers/sensor/bq274xx/bq274xx.c

```diff
@@ -0,0 +1,451 @@
+/*
+ * Copyright (c) 2020 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/i2c.h>
+#include <init.h>
+#include <drivers/sensor.h>
+#include <sys/__assert.h>
+#include <string.h>
+#include <sys/byteorder.h>
+
+#include "bq274xx.h"
+
+/**
+ * @brief sensor value get
+ *
+ * @return -ENOTSUP for unsupported channels
+ */
+static int bq274xx_channel_get(struct device *dev, enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct bq274xx_data *bq274xx = dev->driver_data;
+	float int_temp;
+
+	switch (chan) {
+	case SENSOR_CHAN_GAUGE_VOLTAGE:
+		val->val1 = ((bq274xx->voltage / 1000));
+		val->val2 = ((bq274xx->voltage % 1000) * 1000U);
+		break;
+
+	case SENSOR_CHAN_GAUGE_AVG_CURRENT:
+		val->val1 = ((bq274xx->avg_current / 1000));
+		val->val2 = ((bq274xx->avg_current % 1000) * 1000U);
+		break;
+
+	case SENSOR_CHAN_GAUGE_STDBY_CURRENT:
+		val->val1 = ((bq274xx->stdby_current / 1000));
+		val->val2 = ((bq274xx->stdby_current % 1000) * 1000U);
+		break;
+
+	case SENSOR_CHAN_GAUGE_MAX_LOAD_CURRENT:
+		val->val1 = ((bq274xx->max_load_current / 1000));
+		val->val2 = ((bq274xx->max_load_current % 1000) * 1000U);
+		break;
+
+	case SENSOR_CHAN_GAUGE_TEMP:
+		int_temp = (bq274xx->internal_temperature * 0.1);
+		int_temp = int_temp - 273.15;
+		val->val1 = (s32_t)int_temp;
+		val->val2 = (int_temp - (s32_t)int_temp) * 1000000;
+		break;
+
+	case SENSOR_CHAN_GAUGE_STATE_OF_CHARGE:
+		val->val1 = bq274xx->state_of_charge;
+		val->val2 = 0;
+		break;
+
+	case SENSOR_CHAN_GAUGE_STATE_OF_HEALTH:
+		val->val1 = bq274xx->state_of_health;
+		val->val2 = 0;
+		break;
+
+	case SENSOR_CHAN_GAUGE_FULL_CHARGE_CAPACITY:
+		val->val1 = (bq274xx->full_charge_capacity / 1000);
+		val->val2 = ((bq274xx->full_charge_capacity % 1000) * 1000U);
+		break;
+
+	case SENSOR_CHAN_GAUGE_REMAINING_CHARGE_CAPACITY:
+		val->val1 = (bq274xx->remaining_charge_capacity / 1000);
+		val->val2 = ((bq274xx->remaining_charge_capacity % 1000) * 1000U);
+		break;
+
+	case SENSOR_CHAN_GAUGE_NOM_AVAIL_CAPACITY:
+		val->val1 = (bq274xx->nom_avail_capacity / 1000);
+		val->val2 = ((bq274xx->nom_avail_capacity % 1000) * 1000U);
+		break;
+
+	case SENSOR_CHAN_GAUGE_FULL_AVAIL_CAPACITY:
+		val->val1 = (bq274xx->full_avail_capacity / 1000);
+		val->val2 = ((bq274xx->full_avail_capacity % 1000) * 1000U);
+		break;
+
+	case SENSOR_CHAN_GAUGE_AVG_POWER:
+		val->val1 = (bq274xx->avg_power / 1000);
+		val->val2 = ((bq274xx->avg_power % 1000) * 1000U);
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+
+static int bq274xx_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
```

Comment:
Need to parse `chan` and only read the channel(s) requested

Reply:
Done thanks.